Modules/_struct.c

```diff
@@ -607,9 +607,9 @@ np_ushort(_structmodulestate *state, char *p, PyObject *v, const formatdef *f)
     if (get_long(state, v, &x) < 0)
         return -1;
     if (x < 0 || x > USHRT_MAX) {
-        PyErr_SetString(state->StructError,
-                        "ushort format requires 0 <= number <= "
-                        Py_STRINGIFY(USHRT_MAX));
+        PyErr_Format(state->StructError,
+                     "ushort format requires 0 <= number <= %zu",
```

Comment:
I think we just want `%u` here: `USHRT_MAX` should have type `unsigned int`.

Reply:
Hmm, actually that's not true; for most systems `USHRT_MAX` will have type `int`, so to avoid undefined behaviour we actually want a format specifier of `%d`.  But it's *possible* on unusual systems for `USHRT_MAX` to have type `unsigned int`. (For example, on a system where `short` has the same range as `int`.) We can either (a) worry about those systems when we meet them (which will probably be never), or be extra careful and use `USHRT_MAX + 0U` to force type `unsigned int` and then use `%u`.

Reply:
```suggestion
                     "ushort format requires 0 <= number <= %u",
```